usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -207,36 +207,40 @@ def post(self, request):
         """Return all Parents and Recipients matching the provided search text"""
 
         search_text, limit = self.get_request_payload(request)
-
         # Return Recipients with valid id entries
         queryset = LegalEntity.objects.exclude(
             recipient_unique_id__isnull=True).exclude(
             recipient_unique_id__exact='')
 
         # Filter based on search text
         response = {}
+
         # Search and filter for Parent Recipients
         parents = queryset.annotate(
-            search=SearchVector('recipient_name', 'parent_recipient_unique_id')
-        ).filter(
-            search__icontains=search_text,
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('parent_recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').filter(
             parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
         # Search and filter for Recipients, excluding Parent Recipients
         recipients = queryset.annotate(
```

Comment:
For recipients, how is exact matches handled here? Was something like https://github.com/fedspendingtransparency/usaspending-api/blob/85a84d70cddb6d7e808d26874a19f083a1122724/usaspending_api/references/v2/views/autocomplete.py#L50 not necessary?